tools/c7n_azure/c7n_azure/actions/lock.py

```diff
@@ -64,7 +65,7 @@ def _prepare_processing(self):
         self.client = self.manager.get_client('azure.mgmt.resource.locks.ManagementLockClient')
 
     def _process_resource(self, resource):
-        if resource.get('resourceGroup') is None:
+        if resource['type'] == RESOURCE_GROUPS_TYPE:
```

Comment:
I might have even gone so far as to `resource.is_resource_group()` by adding it to the base :)